.github/workflows/ci-examples.yml

```diff
@@ -16,16 +16,20 @@ env:
   NIGHTLY_TOOLCHAIN: nightly
 
 concurrency:
-  group: ${{github.workflow}}-${{github.ref}}-examples
+  group: ${{github.workflow}}-${{github.event.pull_request.head.sha}}-examples
   cancel-in-progress: ${{github.event_name == 'pull_request_target'}}
 
 jobs:
+  # This job shouldn't use secrets as it checkouts the PR
   run-examples-macos-metal:
     # Explicity use macOS 14 to take advantage of M1 chip.
     runs-on: macos-14
     timeout-minutes: 30
     steps:
+      # Checkout the PR
       - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
How will this act on the `merge_group` or `push` triggers, since `github.event.pull_request` may not be available?

Reply:
yeah I'm having other issues when using this workflow... trying another way